modules/http/http-worker.c

```diff
@@ -760,6 +777,7 @@ _init(LogThreadedDestWorker *s)
   HTTPDestinationDriver *owner = (HTTPDestinationDriver *) self->super.owner;
 
   self->request_body = g_string_sized_new(32768);
+  self->request_body_compressed = g_string_sized_new(32768);
```

Comment:
[optional]

We waste memory here, if compression is not set. Can we skip this in that case?

Reply:
The tradeoffs of wasting memory versus the speedloss of extra conditional executions. If you think the first is a bigger problem than the latter, I will add the `if` clause.

Reply:
It would be great if we had such optimized performance, where an extra `if` in `init()` slowed us down significantly :) However 32K unused memory per worker is a bit more significant IMO.

Reply:
Makes sense. I'm on it

Reply:
Done